### PR TITLE
Manifest references + Node semver migration

### PR DESCRIPTION
**Pull request by simon-id**
## Motivation

Introduces a special optional field `refs` in manifests, it should hold a list of reference definitions, to be used elsewhere in the manifests.
Example:
```yaml
refs:
  - &version_A '>=5.6.0 || 4.7.0'
  - &version_B v2.3.0

tests/:
  file.py:
    ClassA: *version_A
    ClassB: *version_B
```

It synergizes particularly well with SemVer support, to reduce noies and potential errors.

## Changes

- Add documentation on the feature and recommendations on using it.
- Change all the Node.js versions with their full SemVer descriptors and use a ref table for it.
- Add a special case in manifest parsing to not consider the `refs` field as a test file.
- Update the JSON Schema for the manifests.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: manifests/nodejs.yml [hunk 1]
```diff
@@ -1,3 +1,34 @@
+refs:
+  - &ref_3.0.0 '>=3.0.0 || ^2.13.0'
```
Comment (simon-id) [inline manifests/nodejs.yml:2]: @cbeauchesne are you kidding me the YAML library doesn't support non alphanumeric chars in anchors ? xD
```
E   yaml.scanner.ScannerError: while scanning an anchor
E     in "manifests/nodejs.yml", line 2, column 5
E   expected alphabetic or numeric character, but found '.'
E     in "manifests/nodejs.yml", line 2, column 11
```

Comment (simon-id) [reply manifests/nodejs.yml:2]: >YAML anchors and aliases cannot contain the ' [ ', ' ] ', ' { ', ' } ', and ' , ' characters.

So is the yaml parser used currently not following the spec ?

Comment (cbeauchesne) [reply manifests/nodejs.yml:2]: I'm using [the most used yaml parser](https://pyyaml.org/)

Is it something blocking ?

Comment (simon-id) [reply manifests/nodejs.yml:2]: I checked the source code and the only characters allowed are `a-zA-Z0-9-_` which is annoyingly and non-spec-following restrictive. So what would be the alternative ? `&ref_5_6_0`, `&ref_5-6-0` ?

**Comment by simon-id** [comment]
Job is failing because the system-tests transforms the valid semver version provided by the Node library `6.0.0-pre` into `6.0.0pre` which is not a valid string

**Comment by simon-id** [comment]
ok it's fixed now :+1:

**Review by CarlesDD — APPROVED** [review]
LGTM

**Review by uurien — APPROVED** [review]

**Review by cbeauchesne — APPROVED** [review]

**Review by khanayan123 — APPROVED** [review]
LGTM

**Pull request merged**